modules/src/core/ics02_client/client_state.rs

```diff
@@ -40,6 +40,15 @@ pub trait ClientState: Clone + core::fmt::Debug + Send + Sync {
     /// Frozen height of the client
     fn frozen_height(&self) -> Option<Height>;
 
+    /// The client's unbonding period
+    fn unbonding_period(&self) -> Duration;
+
+    /// Helper function to verify the upgrade client procedure.
+    /// Resets all fields except the blockchain-specific ones,
+    /// and updates the given fields.
+    fn upgrade(self, upgrade_height: Height, unbonding_period: Duration, chain_id: ChainId)
```

Comment:
Same comment here regarding `unbonding_period` being specific to Tendermint blockchains. I think we might need a more generic upgrade params abstraction (maybe assoc type?).

Reply:
Fixed in https://github.com/informalsystems/ibc-rs/pull/1625/commits/4b0a78dd082066a18cd11514af8e8b85ec5526f6